development/gcp/cloud-run/create-github-issue/create-github-issue.go

```diff
@@ -0,0 +1,197 @@
+		}
+	}
+
+	logger := cloudfunctions.NewLogger()
+	logger.WithComponent(componentName)
+	logger.WithLabel("io.kyma.app", applicationName)
+	logger.WithLabel("io.kyma.component", componentName)
+	logger.WithTrace(trace)
+
+	requestDump, err := httputil.DumpRequest(r, true)
+	if err != nil {
+		logger.LogError("failed dump http request, error: %s", err.Error())
+	}
+	logger.LogDebug("request:\n%v", string(requestDump))
+
+	event, err := cloudevents.NewEventFromHTTPRequest(r)
+	if err != nil {
+		crhttp.WriteHTTPErrorResponse(w, http.StatusBadRequest, logger, "failed to parse CloudEvent from request: %s", err.Error())
+		return
+	}
+
+	logger.LogInfo("got message, id: %s, type: %s", event.ID(), event.Type())
+
+	// Load event data
+	if err = event.DataAs(&msg); err != nil {
+		crhttp.WriteHTTPErrorResponse(w, http.StatusInternalServerError, logger, "failed marshal event, error: %s", err.Error())
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hasher := sha1.New()
+	hasher.Write([]byte(*msg.JobName + *msg.JobType))
+	h := hasher.Sum(nil)
+	secretsleakscannerID := base64.StdEncoding.EncodeToString(h)
+
+	issueData := templates.SecretsLeakIssueData{
+		SecretsLeaksScannerID:     secretsleakscannerID,
+		ProwMessage:               msg.ProwMessage,
+		SecretsLeakScannerMessage: msg.SecretsLeakScannerMessage,
+	}
+
+	issueBody, err := issueData.RenderBody()
+	if err != nil {
+		logger.LogError("failed render issue body, error: %s", err.Error())
+	}
+
+	issueTitle := fmt.Sprintf("Secret leak found in %s prowjob logs", *msg.JobName)
+	issueLabels := []string{"bug"}
+	issueState := github.String("open")
+	issueRequest := github.IssueRequest{
+		Title:     github.String(issueTitle),
+		Body:      github.String(fmt.Sprint(issueBody.String())),
+		Labels:    &issueLabels,
+		Assignee:  github.String(msg.GithubIssueAssignee.SapToolsGithubUsername),
+		State:     issueState,
+		Milestone: nil,
+		Assignees: nil,
+	}
+	// Search issues
+	issue, response, err := sapGhClient.Issues.Create(ctx, githubOrg, githubRepo, &issueRequest)
+	if err != nil {
+		crhttp.WriteHTTPErrorResponse(w, http.StatusInternalServerError, logger, "failed create github issues, error: %s", err)
+		return
+	}
+	ok, err := kgithubv1.IsStatusOK(response)
+	if err != nil {
+		crhttp.WriteHTTPErrorResponse(w, http.StatusInternalServerError, logger, "failed create github issues, failed read status code, error %s", err)
+		return
+	}
+	if !ok {
+		crhttp.WriteHTTPErrorResponse(w, http.StatusInternalServerError, logger, "failed create github issues, received non 2xx response code: status code %d", response.StatusCode)
+		return
+	}
+	logger.LogInfo("created github issue: %s", issue.GetHTMLURL())
+	msg.GithubIssueNumber = issue.Number
+	msg.GithubIssueURL = issue.HTMLURL
+	msg.GithubIssueOrg = github.String(githubOrg)
+	msg.GithubIssueRepo = github.String(githubRepo)
+	// TODO: Add setting GithubAssigne in msg
```

Comment:
should this TODO comment stay and be merged into main branch?

Reply:
Yes, assigning a user to github issue will be implemented in future.